airflow/hooks/mysql_hook.py

```diff
@@ -114,8 +111,46 @@ def get_conn(self):
             conn_config['unix_socket'] = conn.extra_dejson['unix_socket']
         if local_infile:
             conn_config["local_infile"] = 1
-        conn = MySQLdb.connect(**conn_config)
-        return conn
+        return conn_config
+
+    def _get_conn_config_mysql_connector_python(self, conn):
+        conn_config = {
+            'user': conn.login,
+            'password': conn.password or '',
+            'host': conn.host or 'localhost',
+            'database': self.schema or conn.schema or ''
+        }
+
+        if not conn.port:
+            conn_config['port'] = 3306
+        else:
+            conn_config['port'] = int(conn.port)
```

Comment:
```suggestion
            conn_config['port'] = int(conn.port) if conn.port else 3306
```
Maybe one line, ternary operator?